src/coll/options.rs

```diff
@@ -486,9 +542,11 @@ pub struct AggregateOptions {
     /// information on how to use this option.
     pub collation: Option<Collation>,
 
-    /// Tags the query with an arbitrary string to help trace the operation through the database
-    /// profiler, currentOp and logs.
-    pub comment: Option<String>,
```

Comment:
I think this same change came up in the last `comment` PR: https://github.com/mongodb/mongo-rust-driver/pull/680#discussion_r896959591

There are a few things we can do here (and in other places where this change was made):
- We can wait until we do a 3.0 release of the driver to support `Bson` comments for these commands. At that point we'll be able to make the breaking change and either return an error if a user specifies a non-`Bson::String` pre-4.4 or just propagate the server error that would occur.
- We can introduce separate `comment_bson` options in these structs that accept `Bson` and leave the current `comment` options as-is. At the point of 3.0 we'd probably want to remove them and change the `comment` fields to `Option<Bson>`.

I think I'd prefer the first option given that users haven't been asking for `Bson` comment support here. We can always implement the second option later on if it does turn out to be necessary. In the meantime, users can also use the string representation of their BSON if they do want to use a `Bson` value. (Either way we do this, we'll need to do some workarounds when deserializing the spec tests.)

Reply:
Do you think it would be okay then to revert it to an `Option<String>` and leave a TODO comment to change it to `Bson` before the 3.0 release?

Reply:
Looks like I actually filed a ticket about this when it came up originally, thought I had remembered discussing this a bit in triage: https://jira.mongodb.org/browse/RUST-1364

Yeah, whatever we end up doing we'll need to revert this to an `Option<String>`. Also going to leave this thread open for the rest of the reviewers to weigh in on.